test/ui/CategoryFacet/CategoryFacetSearchTest.ts

```diff
@@ -0,0 +1,191 @@
+import { CategoryFacet } from '../../../src/ui/CategoryFacet/CategoryFacet';
+import { mock, basicComponentSetup } from '../../MockEnvironment';
+import { FakeResults } from '../../Fake';
+import { CategoryFacetSearch } from '../../../src/ui/CategoryFacet/CategoryFacetSearch';
+import { CategoryFacetQueryController } from '../../../src/controllers/CategoryFacetQueryController';
+import _ = require('underscore');
+import { IGroupByValue } from '../../../src/rest/GroupByValue';
+import { $$ } from '../../../src/Core';
+import { KEYBOARD } from '../../../src/utils/KeyboardUtils';
+
+export function CategoryFacetSearchTest() {
+  describe('CategoryFacetSearch', () => {
+    let categoryFacetMock: CategoryFacet;
+    let categoryFacetSearch: CategoryFacetSearch;
+    let realDebounce;
+    let fakeGroupByValues: IGroupByValue[];
+    beforeEach(() => {
+      realDebounce = _.debounce;
+      spyOn(_, 'debounce').and.callFake(function(func) {
+        return function() {
+          func.apply(this, arguments);
+        };
+      });
+      categoryFacetMock = basicComponentSetup<CategoryFacet>(CategoryFacet).cmp;
+      fakeGroupByValues = FakeResults.createFakeGroupByResult('@field', 'value', 10).values;
+      categoryFacetMock.categoryFacetQueryController = mock(CategoryFacetQueryController);
+      categoryFacetMock.categoryFacetQueryController.searchFacetValues = () => new Promise(resolve => resolve(fakeGroupByValues));
+      categoryFacetSearch = new CategoryFacetSearch(categoryFacetMock);
+    });
+
+    afterEach(() => {
+      _.debounce = realDebounce;
+    });
+
+    it('when building returns a container', () => {
+      const container = categoryFacetSearch.build();
+      expect(container.hasClass('coveo-category-facet-search-container')).toBe(true);
+    });
+
+    it('focus moves the focus to the input element', () => {
+      categoryFacetSearch.facetSearchElement = { input: jasmine.createSpyObj('input', ['focus']) } as any;
+      spyOn(categoryFacetSearch.facetSearchElement, 'input');
+
+      categoryFacetSearch.focus();
+
+      expect(categoryFacetSearch.facetSearchElement.input.focus).toHaveBeenCalled();
```

Comment:
We could add utilities function in those tests to access the elements needed, instead of having to re-type everytime 
`categoryFacetSearch.facetSearchElement.input` or `categoryFacetSearch.facetSearchElement.searchResults`